### PR TITLE
Detect negated content absent buffer 2224 v10

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2224
https://redmine.openinfosecfoundation.org/issues/6629

Describe changes:
- detect: negated content matches on absent buffer
- detect: adds `absent` keyword to match on absent buffer
- doc: fix `byte_test`examples

```
SV_BRANCH=pr/1535
```
https://github.com/OISF/suricata-verify/pull/1535

#10034 with
- ticket reference for doc fixes. Should the commit go into its own PR ?
- code review taken into account

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 181 | 195 | 107.73% | 

Pipeline 17063

File: doc/userguide/rules/payload-keywords.rst [hunk 1]
```diff
@@ -278,6 +278,18 @@ A negated isdataat on an absent buffer will match any value.
```
Comment (victorjulien) [inline doc/userguide/rules/payload-keywords.rst:278]: please explain both idea and high level implementation in the commit message

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10140

File: doc/userguide/rules/payload-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/payload-keywords.rst:278]: ok trying to improve the commit message

**Pull request closed without merging**